src/signature.rs

```diff
@@ -149,7 +149,7 @@ mod tests {
         }
     }
 
-    fn gen_n_pubkeys(seed: [u8; 32], n: i64) -> HashSet<PublicKey> {
+    fn gen_n_pubkeys(seed: [u8; 32], n: i64) -> HashSet<Pubkey> {
```

Comment:
generate? We use `gen_` a few ways, not all of our making. But we use `generate_` too...

Reply:
@CriesofCarrots, I missed this comment. Yes, I can fix that up in a separate PR. I can also rename GenKeys to KeyGenerator while I'm in the neighborhood.